src/jit/lclvars.cpp

```diff
@@ -108,12 +108,19 @@ void Compiler::lvaInitTypeRef()
     /* Set compArgsCount and compLocalsCount */
 
     info.compArgsCount = info.compMethodInfo->args.numArgs;
+    unsigned   argRegCount      = 0;
+
+    auto incrementArgCount = [this, &argRegCount]()
+    {
+        ++info.compArgsCount;
+        ++argRegCount;
+    };
 
```

Comment:
This code assumes that anything using incrementArgCount (which is the four specials: 'this', retbuf, generics context, varargs cookie) are register arguments. But that doesn't appear to be true, e.g., for x86, which only has 2 register arguments. (There are also "non-standard" reg args for things like pinvoke helpers, etc.).

It seems like the only code that can count reg args correctly is fgMorphArgs, when AddRegArg() is called. I wonder if this calculation should be done then instead.